os/binfmt/binfmt_loadbinary.c

```diff
@@ -204,12 +208,28 @@ int load_binary(int binary_idx, FAR const char *filename, load_attr_t *load_attr
 			goto errout_with_unload;
 		}
 
-		return pid;
 #ifdef CONFIG_SUPPORT_COMMON_BINARY
 	}
+#endif
+
+#ifdef CONFIG_DEBUG_BIN_SYMBOL
+	bin_addr_info_t *bin_info;
+	bin_info = (bin_addr_info_t *)kmm_malloc(sizeof(bin_addr_info_t));
+	if (bin_info != NULL) {
```

Comment:
If anything stopped because of return value, then you'd better follow below style to improve redability.
That is always easy to understand
```
if (a != b) {
    return or  goto
}
do_something;
```
Of course depth is just one here, you can ignore this comment.

Reply:
This is just debug option, not for module operation. So even if not allocated properly, I don't want to stop the operation even if this debug functionality fails.

Reply:
but it just return pid.

Reply:
If no mem, there is no allocated resources from debug feature, and it just print the error msg.